### DISCUSSION TITLE
SlashCommandStringOption.addChoices() quesiton

### DISCUSSION THREAD

**Discussion by Mirai-Miki** [discordjs/discord.js#7825] [Q&A] [answered]
I just updated to @discordjs/builders v0.13.0 and am having a few issues with some of the changes.
In the old version to add string options to a SlashCommandStringOption you would just set them with .addChoice(). That doesn't exist anymore in 0.13 and instead has been replaced with .addChoices() and .setChoices().

The problem I have having is the documentation isn't very clear on what should be passed to these as arguments.
![Capture](https://user-images.githubusercontent.com/40791806/164625871-dc3c265e-543b-48c7-9d1f-6c09b90ced03.PNG)

I cannot seem to figure out what an APIApplicationCommandOptionChoice is. I figured it was just an array of objects containing the name and value, as that is what the discord API is expecting.

```js
[
   {name: "text", value: "text"},
   {name: "text", value: "text"}
]
```
But that is thowing a CombinedPropertyError at ArrayValidator.parse so I am really not sure what needs to be put there.

I also had a look at the .setChoices() method but that has even worse documentation then the other one.
![Capture](https://user-images.githubusercontent.com/40791806/164626818-9ab90c1b-350a-40f3-a29c-8dc98fa037c8.PNG)
No description at all and who knows what is expected at ...Input?

The discord.js guide is no help either as it is still using the old method of adding choices.

Anyone know how to add choices for a string command using this version of builders?

**Comment by vladfrangu** [comment, marked as answer]
In that version, just pass in an object to addChoices (`.addChoices({ name, value })`), but in a future version the array you mentioned will be the correct way of using the method

**Comment by monbrey** [comment]
This is very bad timing to be answering this but here we go...

In the current 0.13.0 release of builders the add/set methods take rest parameters. That is, as many objects as you like NOT in an array, separated by commas.

The next release, and currently in main/dev release, it has been changed to an array (the way you attempted) because... well, it's the way people naturally attempt to and expect to use these methods.

**Reply by mrcordova** [reply]
Any plans to allow multiple choice selection with setChoices like in select menus?

**Reply by almostSouji** [reply]
This is a question for Discord. We can just implement what [the API allows](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure).

For suggestions and feedback regarding the API, see https://github.com/discord/discord-api-docs/discussions

**Reply by mrcordova** [reply]
Understood, thank you for letting me know.

**Reply by FelipeTolentino** [reply]
Still not changed? I'm currently working on version 14.13.0 and looking in the documentation I found the same as Mirai-Miki, but when I tried to pass an array to .addChoices() I get the error 'ValidationError: Expected the value to be not an array'. 

Thanks in advance

**Reply by Sowgro** [reply]
I have run into the same issue. Placing `...` before the name of the array worked for me
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#A_better_apply
Also note there appears to be a limit of 25.
